src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/services/OffenceCodes.kt

```diff
@@ -102,5 +103,22 @@ enum class OffenceCodes(val paragraph: String, private val withOthers: String? =
 
     fun validateOffenceCode(offenceCode: Int) =
       OffenceCodes.values().flatMap { it.uniqueOffenceCodes }.firstOrNull { it == offenceCode } ?: throw ValidationException("Invalid offence code $offenceCode")
+
+    fun findByNomisCode(nomisCode: String): Int {
```

Comment:
a wip, will use this code (and exceptions) to pull out the mappings that are a problem.